app.py

```diff
@@ -39,43 +40,55 @@
 
 # Setup Logging
 logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"),
-                    format='%(levelname)s: %(message)s')
+                    format='[%(asctime)s] %(levelname)s: %(message)s',
+                    datefmt='%m/%d/%y %H:%M:%S')
 
 # Global variables
 static_img_path = "static/img/images/"
 temp_img_prefix = "MAX-"
 image_captions = collections.OrderedDict()
-VALID_EXT = ['png', 'jpg', 'jpeg']
+VALID_EXT = ['.png', '.jpg', '.jpeg']
 error_raised = []
+app_cookie = 'max-image-caption-generator-web-app'
 
 
-class MainHandler(web.RequestHandler):
+class BaseHandler(web.RequestHandler):
+    def prepare(self):
+        if not self.get_cookie(app_cookie):
```

Comment:
might be worth adding a link to http://www.tornadoweb.org/en/branch2.4/web.html#cookies as a comment here

Reply:
Will do

Reply:
Using the Doc link instead: http://www.tornadoweb.org/en/stable/guide/security.html